### PR TITLE
Highlight matched text when adding Contact to an email

### PR DESCRIPTION
**Pull request by LunarX**
Depends on #829

### CODE REVIEW

File: app/src/main/java/com/infomaniak/mail/ui/main/AvatarNameEmailView.kt [hunk 1]
```diff
@@ -85,6 +92,21 @@ class AvatarNameEmailView @JvmOverloads constructor(
         binding.root.setOnClickListener(onClickListener)
     }
 
+    fun highlight(nameStartIndex: Int, emailStartIndex: Int, length: Int) = with(binding) {
+        if (nameStartIndex > -1) userName.highlight(nameStartIndex, nameStartIndex + length)
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/ui/main/AvatarNameEmailView.kt:96]: Magic value.

Did you mean `NOT_SET` ?

File: app/src/main/java/com/infomaniak/mail/ui/main/AvatarNameEmailView.kt [hunk 2]
```diff
@@ -85,6 +92,21 @@ class AvatarNameEmailView @JvmOverloads constructor(
...
+        if (emailStartIndex > -1 && userEmail.text.isNotBlank()) userEmail.highlight(emailStartIndex, emailStartIndex + length)
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/ui/main/AvatarNameEmailView.kt:97]: Magic value.

Did you mean `NOT_SET` ?

Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/ui/main/AvatarNameEmailView.kt:97]: The `userEmail` can be blank, but not the `userName` ? Isn't it the opposite ?

File: app/src/main/java/com/infomaniak/mail/ui/main/newMessage/ContactAdapter.kt [hunk 3]
```diff
@@ -79,55 +80,50 @@ class ContactAdapter(
...
-                for (contact in allContacts) {
-                    val standardizedEmail = contact.email.standardize()
-                    val matches = contact.name.standardize().contains(searchTerm) || standardizedEmail.contains(searchTerm)
+            val finalUserList = mutableListOf<MatchedContact>()
+            displayAddUnknownContactButton = true
+            for (contact in allContacts) {
+                val nameMatchedIndex = contact.name.standardize().indexOf(searchTerm)
+                val standardizedEmail = contact.email.standardize()
+                val emailMatchedIndex = standardizedEmail.indexOf(searchTerm)
+                val matches = nameMatchedIndex > -1 || emailMatchedIndex > -1
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/ui/main/newMessage/ContactAdapter.kt:107]: Magic value twice.

Did you mean `NOT_SET` ?

**Review by KevinBoulongne — CHANGES_REQUESTED** [review]

**Review by KevinBoulongne — APPROVED** [review]

**Review by FabianDevel — CHANGES_REQUESTED** [review]

**Review by FabianDevel — APPROVED** [review]

File: app/src/main/java/com/infomaniak/mail/ui/main/newMessage/ContactAdapter.kt [hunk 4]
```diff
@@ -80,55 +81,51 @@ class ContactAdapter(
         }
     }
 
-    override fun getItemCount(): Int = contacts.count() + if (displayAddUnknownContactButton) 1 else 0
+    override fun getItemCount(): Int = matchedContacts.count() + if (displayAddUnknownContactButton) 1 else 0
 
-    override fun getItemId(position: Int) = if (getItemViewType(position) == KNOWN_CONTACT.id) contacts[position].id!! else 0
+    override fun getItemId(position: Int): Long {
+        return if (getItemViewType(position) == KNOWN_CONTACT.id) matchedContacts[position].contact.id!! else 0
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/ui/main/newMessage/ContactAdapter.kt:87]:
```suggestion
        return if (getItemViewType(position) == KNOWN_CONTACT.id) matchedContacts[position].contact.id!! else 0L
```

**Pull request merged**